### PR TITLE
Mobile Nav Menu Fixes

### PR DESCRIPTION
**Pull request by tr3mulant**
Summary of Changes

- Better alignment on newsletter container error/success text
- handleClick() for mobile menu
    - toggles the state openMenu
    - toggles a class on body tag to hide overflow to prevent body from scrolling when the menu is open
- Scaled down the NavBrand size on small screens
- Small resize on NewsletterSubscribe embed on mobile menu
- Adjustment to height of fixed mobile menu when opened to offset the top position
- Structured the home page to use grid layout
    - Added a header tag and main tag
- Initial add for 404 and 500
- Github workflows divided up as a build/test and deploy to hopefully have a better flow

### CODE REVIEW

File: components/NavBrand.js [hunk 1]
```diff
@@ -13,7 +13,8 @@ const NavBrandContainer = styled.div`
         margin-left: auto;
     }
     @media only screen and (max-width: ${(props) => props.theme.breakpoints.sm}) {
-        margin-left: ${(props) => props.theme.spaces.md}
+        margin-left: ${(props) => props.theme.spaces.md};
+        width: ${(props) => props.theme.spaces.xxxl.replace("rem","") * 0.7}rem;
```
Comment (calebn) [inline components/NavBrand.js]: This is fine but I also don't think there's anything wrong with using a hard coded width number for elements like this. I think of spacing more as used for margin/padding etc.

File: components/Navbar.js [hunk 2]
```diff
@@ -63,17 +67,26 @@ const NavLinks = ({ openMenu, components, onClick }) => {
 
 export function Navbar() {
     const [openMenu, toggleMenu] = useState(false);
+    const handleClick = (state) => {
+        //feels hacky, like maybe the state should be managed higher up?
```
Comment (calebn) [inline components/Navbar.js:71]: I was just reading an [article about the framer-motion page transitions](https://wallis.dev/blog/nextjs-page-transitions-with-framer-motion) that I think had a handle for this. Scroll down to the part titled "Solving the scroll on link change issue." They recommended creating a link component called `NoScrollLink` that uses the [scroll="false"](https://nextjs.org/docs/api-reference/next/link#disable-scrolling-to-the-top-of-the-page) attribute.

Would that work here?

**Review by calebn — COMMENTED** [review]
Take a look at that scroll="false" property and if it seems like it could work I'd recommend implementing that.

File: components/NavBrand.js (hunk 1 above)
Comment (tr3mulant) [reply components/NavBrand.js]: Fair enough.
Will change line 17 to:
`width: 6.72rem;`

File: components/Navbar.js (hunk 2 above)
Comment (tr3mulant) [reply components/Navbar.js:71]: I'll read that and report back tonight or tomorrow

Comment (tr3mulant) [reply components/Navbar.js:71]: I don't think the handleClick method is tackling the same problem as the the NoScrollLink component is tackling.

The NoScrollLink component seems to be handling the on click event which scrolls to the top when the route is changed and then triggers the animation. This causes the awkward motion and stutter/judder and feels janky. I do believe this is a useful component regardless and is probably a good idea to implement the component and have our styled navigation links extend the NoScrollLink component.

The method in question is preventing scrolling on the background elements (header, main, footer) on mobile when the mobile menu is open. If a user were to open a menu and scroll down the navigation links to see the newsletter signup and decided to close the menu, they would be in a different position on the page as they would be scrolling both the open mobile menu and the page itself.

In hindsight, not even sure how big of deal it is

**Review by calebn — APPROVED** [review]

**Pull request merged**